x/ccv/provider/keeper/distribution.go

```diff
@@ -150,7 +150,7 @@ func (k Keeper) AllocateTokensToConsumerValidators(
 		consAddr := sdk.ConsAddress(consumerVal.ProviderConsAddr)
 
 		// get the validator tokens fraction using its voting power
-		powerFraction := math.LegacyNewDec(consumerVal.Power).QuoTruncate(math.LegacyNewDec(totalPower))
+		powerFraction := math.LegacyNewDec(consumerVal.Power).QuoTruncate(totalPower)
```

Comment:
Changed due to the "math.LegacyNewDec(totalPower) should be initialized once before entering the loop" suggestion.